core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -42,7 +42,7 @@ import scala.concurrent.ExecutionContext
  * values, see the default interpreter in `RTS` or the safe main function in
  * `App`.
  */
-sealed abstract class IO[+E, +A] { self =>
+sealed abstract class IO[+E, +A] extends Serializable { self =>
```

Comment:
[With Product](https://typelevel.org/blog/2018/05/09/product-with-serializable.html), here and elsewhere.

Reply:
Can make the IO extend Product but the materializations of IO are classes and not case classes hence we need to implement the abstract methods of ```Product``` inside, or make these case classes:

```final class Suspend[E, A] private[IO] (val value: () => IO[E, A]) extends IO[E, A]```

Reply:
You are absolutely right, there's no need to extend `Product` for `IO`. For other types which are case classes, however, let's try to do that.